### DISCUSSION TITLE
Potential Internal Update to Lua 5.4

### DISCUSSION THREAD

**Discussion by orbitalquark** [orbitalquark/textadept#56] [General]
Would anyone be terribly inconvenienced if Textadept upgraded its internal copy of Lua to 5.4? The only real issue I see is for external modules that depend on external, binary libraries (e.g. LuaSocket). Anyone depending on a binary library compiled for Lua 5.3 would need to use a version compiled for Lua 5.4. Then again, LuaRocks can probably help ease any pain there.

I don't really know what the state of 3rd-party Textadept modules depending on external, binary Lua modules is. I'd imagine there's not much going on since Textadept is just a text editor, but I thought I'd ask.

FYI I am updating my debugger module to bundle a local copy of LuaSocket for use by Textadept to debug Lua scripts. (LuaSocket is still required for the external Lua interpreter invoked.) My local copy of a Lua 5.4-enabled Textadept can debug a Lua 5.3 script using the system's LuaSocket compiled for Lua 5.3, so updating to 5.4 would not be an issue for me.

**Comment by paaguti** [comment]
I don't know if that will impact the YAML module... hopefully not.

I'm testing when cycles available :-)

Best, /PA

On Mon, 15 Feb 2021 at 02:45, orbitalquark <notifications@github.com> wrote:

> Would anyone be terribly inconvenienced if Textadept upgraded its internal
> copy of Lua to 5.4? The only real issue I see is for external modules that
> depend on external, binary libraries (e.g. LuaSocket). Anyone depending on
> a binary library compiled for Lua 5.3 would need to use a version compiled
> for Lua 5.4. Then again, LuaRocks can probably help ease any pain there.
>
> I don't really know what the state of 3rd-party Textadept modules
> depending on external, binary Lua modules is. I'd imagine there's not much
> going on since Textadept is just a text editor, but I thought I'd ask.
>
> FYI I am updating my debugger module to bundle a local copy of LuaSocket
> for use by Textadept to debug Lua scripts. (LuaSocket is still be required
> for the external Lua interpreter invoked.) My local copy of a Lua
> 5.4-enabled Textadept can debug a Lua 5.3 script using the system's
> LuaSocket compiled for Lua 5.3, so updating to 5.4 would not be an issue
> for me.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/56>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4AVBVSMJJT2IS3NAE3S7B4EFANCNFSM4XT2TI3Q>
> .
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

**Comment by orbitalquark** [comment]
The YAML module works fine for me on my local copy of Textadept with Lua 5.4.

**Comment by paaguti** [comment]
OK :-)

On Mon, 15 Feb 2021 at 14:52, orbitalquark <notifications@github.com> wrote:

> The YAML module works fine for me on my local copy of Textadept with Lua
> 5.4.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/56#discussioncomment-370566>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4DUSV7D7V26UGGKYALS7ERK7ANCNFSM4XT2TI3Q>
> .
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler
